Build failure: pkgsLLVM.libbsd

### Steps To Reproduce

Steps to reproduce the behavior:
1. build `pkgsLLVM.libbsd`

### Build log

```
24 warnings generated.
  CCLD     libbsd.la
aarch64-unknown-linux-gnu-ld: error: version script assignment of 'LIBBSD_0.10.0' to symbol 'vwarn' failed: symbol not defined
aarch64-unknown-linux-gnu-ld: error: version script assignment of 'LIBBSD_0.10.0' to symbol 'vwarnx' failed: symbol not defined
aarch64-unknown-linux-gnu-ld: error: version script assignment of 'LIBBSD_0.10.0' to symbol 'warn' failed: symbol not defined
aarch64-unknown-linux-gnu-ld: error: version script assignment of 'LIBBSD_0.10.0' to symbol 'warnx' failed: symbol not defined
aarch64-unknown-linux-gnu-ld: error: version script assignment of 'LIBBSD_0.10.0' to symbol 'verr' failed: symbol not defined
aarch64-unknown-linux-gnu-ld: error: version script assignment of 'LIBBSD_0.10.0' to symbol 'verrx' failed: symbol not defined
aarch64-unknown-linux-gnu-ld: error: version script assignment of 'LIBBSD_0.10.0' to symbol 'err' failed: symbol not defined
aarch64-unknown-linux-gnu-ld: error: version script assignment of 'LIBBSD_0.10.0' to symbol 'errx' failed: symbol not defined
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [Makefile:651: libbsd.la] Error 1
make[2]: Leaving directory '/build/libbsd-0.11.8/src'
make[1]: *** [Makefile:421: all-recursive] Error 1
make[1]: Leaving directory '/build/libbsd-0.11.8'
make: *** [Makefile:353: all] Error 2
```

Looks like a version script error similar to other packages, should be a relatively easy fix.

### Additional context

Add any other context about the problem here.

### Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"aarch64-linux"`
 - host os: `Linux 5.15.158, NixOS, 24.05 (Uakari), 24.05.20240506.6ff120a`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - nixpkgs: `/run/current-system/nixpkgs`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/libraries/libbsd/default.nix
+++ b/pkgs/development/libraries/libbsd/default.nix
@@ -35,6 +35,9 @@ stdenv.mkDerivation rec {
     url = "https://gitlab.freedesktop.org/libbsd/libbsd.git";
   };
 
+  # Fix undefined reference errors with version script under LLVM.
+  configureFlags = lib.optionals (stdenv.cc.bintools.isLLVM && lib.versionAtLeast stdenv.cc.bintools.version "17") [ "LDFLAGS=-Wl,--undefined-version" ];
+
   meta = with lib; {
     description = "Common functions found on BSD systems";
     homepage = "https://libbsd.freedesktop.org/";

